### ISSUE TITLE
st2 run packs.install should register all pack components by default

### ISSUE THREAD

**Issue by armab** [StackStorm/st2#1619]
It seems related to recent: `Action aliases are registered by default. (improvement)` change.

The problem I'm experiencing in chatops is that not everything loaded/unloaded when running `install`/`uninstall` for packs. It's just nice to have `install`/`uninstall` process complete, so user shouldn't care about low-level details.

My behavior: 

* `st2 run packs.install packs=hubot`
* install, configure and run hubot with `hubot-stackstorm`
* expect that I get command output from st2 in chat `!pack info hubot `
* because `chatops.notify_hubot` rule shipped with `hubot` pack wasn't registered - I don't get any output in chat

I think `install` could do all the things by default, not only [register `actions`](https://github.com/StackStorm/st2/blob/68607959047321031d7a3214c75e8eb7fb94dc40/contrib/packs/actions/install.meta.yaml#L24). Maybe I'm missing some use cases here when only `actions` should be registered?

------
Related problem for uninstall. It could be better if that command unload all pack contents as well.

```sh
st2 run packs.uninstall packs=hubot
...
```
After uninstall there is still rule registered:
```sh
st2 rule list -p hubot
+----------------------------+-------+----------------------+------------------------------------------+
| ref                        | pack  | name                 | description                              |
+----------------------------+-------+----------------------+------------------------------------------+
| hubot.chatops.notify_hubot | hubot | chatops.notify_hubot | Notification rule to send messages to    |
|                            |       |                      | Hubot                                    |
+----------------------------+-------+----------------------+------------------------------------------+

```

-----

Third problem, 
seems action-aliases are still not registered:
```sh
st2 run packs.install packs=st2-google repo_url=jfryman/st2-google
...
st2 action-alias list | grep google
# no action aliases registered
```
These commits might be related: [[ 1 ]](https://github.com/StackStorm/st2/commit/a22b3e9c058569aebeb3236fcae257cb52f5c94a) [[ 2 ]](https://github.com/StackStorm/st2/commit/4ceca01035cc86f8a399acabc47b10a27222fdc3)

-----
^^ Tested on `latest` version.
```sh
st2 --version
st2 ('0.12dev.31',)
```

If that makes sense I can submit PR and verify & test everything again. My intent is to install/uninstall st2 packs in chatops way without any additional steps.

**Comment by manasdk** [comment]
We have actually decided not to install rules by default from a pack. Rules are install specific and is safer for the administrator to opt-in for rule installation. How about we update instructions in hubot to use `all` as the value for the `packs.install:register` property (https://github.com/StackStorm/st2/blob/master/contrib/packs/actions/install.meta.yaml#L22). If that is broke too feel free to open a PR that fixes this part.

however install aliases by default should work. I am :disappointed: it did not work for you.

Relevant changes -
https://github.com/StackStorm/st2/commit/c2d22599ba2789c5a87febe841eb52424df454e8
https://github.com/StackStorm/st2/commit/5dce14fbfcad92f3f82ddfb169f04fbeb6673c8b

**Comment by armab** [comment]
> however install aliases by default should work. I am :disappointed: it did not work for you.

```sh
# st2 run packs.install packs=st2-google repo_url=jfryman/st2-google
.................
id: 5579c31f0a84b4401da7efc1
action.ref: packs.install
status: succeeded
start_timestamp: 2015-06-11T17:19:27.909493Z
end_timestamp: 2015-06-11T17:20:01.971882Z
+--------------------------+-----------+--------------------------+-------------------------+----------------------------+
| id                       | status    | task                     | action                  | start_timestamp            |
+--------------------------+-----------+--------------------------+-------------------------+----------------------------+
| 5579c3200a84b44013731216 | succeeded | download                 | packs.download          | Thu, 11 Jun 2015 17:19:28  |
|                          |           |                          |                         | UTC                        |
| 5579c32f0a84b44013731218 | succeeded | virtualenv_prerun        | packs.virtualenv_prerun | Thu, 11 Jun 2015 17:19:43  |
|                          |           |                          |                         | UTC                        |
| 5579c3310a84b4401373121a | succeeded | setup_virtualenv         | packs.setup_virtualenv  | Thu, 11 Jun 2015 17:19:45  |
|                          |           |                          |                         | UTC                        |
| 5579c3390a84b4401373121c | succeeded | reload                   | packs.load              | Thu, 11 Jun 2015 17:19:53  |
|                          |           |                          |                         | UTC                        |
| 5579c33d0a84b4401373121e | succeeded | restart-sensor-container | packs.restart_component | Thu, 11 Jun 2015 17:19:57  |
|                          |           |                          |                         | UTC                        |
+--------------------------+-----------+--------------------------+-------------------------+----------------------------+


# st2 action-alias list | grep google
# nothing here.... <---


# st2ctl reload
Registering content...[flags = --register-sensors --register-actions --register-aliases]
2015-06-11 17:36:14,762 INFO [-] Connecting to database "st2" @ "0.0.0.0:27017" as user "None".
2015-06-11 17:36:14,837 INFO [-] =========================================================
2015-06-11 17:36:14,838 INFO [-] ############## Registering sensors ######################
2015-06-11 17:36:14,839 INFO [-] =========================================================
2015-06-11 17:36:15,171 INFO [-] Registered 1 sensors.
2015-06-11 17:36:15,172 INFO [-] =========================================================
2015-06-11 17:36:15,173 INFO [-] ############## Registering actions ######################
2015-06-11 17:36:15,173 INFO [-] =========================================================
2015-06-11 17:36:16,610 INFO [-] Registered 43 actions.
2015-06-11 17:36:16,612 INFO [-] =========================================================
2015-06-11 17:36:16,614 INFO [-] ############## Registering aliases ######################
2015-06-11 17:36:16,614 INFO [-] =========================================================
2015-06-11 17:36:16,661 INFO [-] ActionAlias /opt/stackstorm/packs/st2-google/aliases/google.yaml not found. Creating new one.
2015-06-11 17:36:16,682 INFO [-] Registered 6 aliases.
##### st2 components status #####
st2actionrunner PID: 16402
st2actionrunner PID: 16403
st2actionrunner PID: 16404
st2actionrunner PID: 16405
st2actionrunner PID: 16406
st2actionrunner PID: 16407
st2actionrunner PID: 16408
st2actionrunner PID: 16409
st2actionrunner PID: 16410
st2actionrunner PID: 16411
st2api PID: 16413
st2auth PID: 16412
st2sensorcontainer PID: 17337
st2rulesengine PID: 16417
mistral PID: 16424
mistral PID: 16465
mistral PID: 16466
st2web PID: 16429
st2resultstracker PID: 16415
st2notifier PID: 16416


# st2 action-alias list | grep google
| st2-google.google_query | st2-google | google_query  |                                         | True    |
# ^^ here it is!
```

They appeared only after `st2ctl reload` (not after `packs.install`).

**Comment by armab** [comment]
Take a look: http://i.imgur.com/MX7wHm3.png Only `actions` are reloaded.

**Comment by manasdk** [comment]
> # st2 run packs.install packs=st2-google repo_url=jfryman/st2-google

Make sense. That is because with packs.install we use register=actions by default. There could be some improvements here to avoid passing in any value and therefore registering `actions`, `sensors` and `action-alias` i.e. preserve the `st2ctl reload` default behavior. 

That fix would have to be in `packs.load` - https://github.com/StackStorm/st2/blob/master/contrib/packs/actions/load.yaml#L12

I suspect some jinja templating magic would work. Not a jinja Ninja myself.

**Comment by manasdk** [comment]
@armab Feel free to submit a PR if you are motivated :)

**Comment by armab** [comment]
^^ looks like better approach, @manasdk take a look

**Comment by armab** [comment]
Shouldn't `st2ctl clean` register all components by default after dropping all databases and logs?
For the moment It uses default flags for `reload`, eg: [`--register-actions --register-aliases --register-sensors`](https://github.com/StackStorm/st2/blob/36af04f2caa03b396fb8ab00fd6d700e827fda8d/tools/st2ctl#L98).

It means after running `st2ctl clean` user gets system without registered `rules` and `sensors` which is confusing IMO if they worked before.
